_posts/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm.md

```diff
@@ -0,0 +1,247 @@
+---
+author: Dan Cohen
+date: 2017-11-13 08:00:00 +00:00
+layout: post
+asset-type: post
+slug: tetris-ai-single-parent-evolutionary-algorithm
+title: "Tetris AI, Experiments 1 & 2: Single Parent Evolutionary Algorithm"
+canonical:
+    name: hacking dan
+    href: http://blog.hackingdan.com/2017/11/tetris-ai-experiment-1-single-parent.html
+image:
+    src: /assets/img/custom/blog/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm/neural-network.png
+tags:
+- evolutionary algorithm
+- machine learning
+- artificial intelligence
+- neural network
+---
+
+## Abstract (why not make it formal)
+
+In this experiment, I implement an evolutionary algorithm with no crossover to evolve a neural network with the intention of having it learn to play tetris. I assess the results of my evolutionary algorithm applied to learning tetris. The results show the impact of a high occurrence of mutations(χ) and a high standard deviation for mutations(σ)  
+
+All the code is available [on github](https://github.com/cohen990/evolution-of-tetris/tree/3147c87c84aafce0d3ed21d9b21b0b0ebae92a30)
+
+## Introduction
+
+### Why?
+
+I have been experimenting with artificial intelligence. I was looking for a game. Ideally something with visual output so that I could see the results of my experiments visually. I was inspired to do this by a series of videos I found on youtube about using an evolutionary algorithm to train simple walkers [available here](https://www.youtube.com/watch?v=GOFws_hhZs8&ab_channel=carykh). I have previously looked into AI [here](https://github.com/cohen990/NeuralNetworkTutorial) and [here](https://github.com/cohen990/Coursera); this is the first time I've seen any notable results.
+
+### Theoretical Background
+
+#### Neural Network
+
+Neural networks are used as an analogy for the decision making pathways in a brain.
+
+[![diagram of fully connected neural network]({{site.baseurl}}/assets/img/custom/blog/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm/neural-network.png)]({{site.baseurl}}/assets/img/custom/blog/2017-11-13-tetris-ai-experiment-1-2-single-parent-evolutionary-algorithm/neural-network.png)
+
+It is structured in three layers. The input layer, the hidden layer and the output layer. In a fully connected neural network, every node in a layer is connected to every node in the following layer. Each node holds a value which is calculated from the previous layer.
```

Comment:
Would it be interesting to mention that the number of hidden layers can be greater than 1 for other neural networks architectures?